app/questionnaire/router.py

```diff
@@ -270,15 +271,31 @@ def _is_section_enabled(self, section):
         if "enabled" not in section:
             return True
 
-        for condition in section["enabled"]:
-            if evaluate_when_rules(
-                condition["when"],
+        enabled = section["enabled"]
+        if isinstance(enabled, Sequence):
+            for condition in enabled:
+                if evaluate_when_rules(
+                    condition["when"],
+                    self._schema,
+                    self._metadata,
+                    self._answer_store,
+                    self._list_store,
+                ):
+                    return True
+        else:
+            location = Location(section_id=section["id"])
```

Comment:
I don't follow why this has been done?.

Reply:
So I removed it from here and made `location` optional in `WhenRuleEvaluator` but that lead to new line of code, assigning `list_item_id` a conditional value as it is part of Location class (and that is done in some other place in runner as well):
`list_item_id = self.location.list_item_id if self.location else None`